spaceros.repos

```diff
@@ -13,3 +13,7 @@ repositories:
     type: git
     url: https://github.com/ament/ament_lint.git
     version: spaceros
+  iceoryx:
+    type: git
+    url: https://github.com/eclipse-iceoryx/iceoryx.git
+    version: v2.0.2
```

Comment:
I haven't figured out if there's a better way to handle this.

"this" being the fact that rosinstall_generator basically expects that the tag name when using the `--upstream` flag is the verbatim version of the repository. The ROS community overall follows this convention but other common conventions like prefixing the version tag with `v` also exist and are used by some projects, particularly ones that have uses outside ROS. Bloom can deal with this by using a template for mapping the package version to a vcs tag but rosinstall_generator can't do that without reading information from the release repository.

In order to resolve this issue for iceoryx, I've added it to the list of Space ROS overrides. Which isn't ideal as it means that this package will need to be manually audited when updating packages to the latest humble release.

Reply:
Interesting. Thanks for the explanation.